Ix.NET/Source/System.Linq.Async/System/Linq/Operators/ToObservable.cs

```
@@ -32,30 +31,35 @@ public IDisposable Subscribe(IObserver<T> observer)
 
                 async void Core()
                 {
-                    try
+                    // REVIEW: fire-and-forget DisposeAsync?
+                    await using (var e = _source.GetAsyncEnumerator(ctd.Token))
```

`e.DisposeAsync()` can potentially throw without being catched.

---

I think is is ok, it shouldn't throw since we you are now adhering to the correct flow, never cancelling in-flight. If it does anyway, it's really exceptional and there's nothing reasonable that can be done anyway I guess.

---

Well, we could forward the exception to the observer, instead of swallowing it.

---

I didn't mean swallow it, although effectively it is what will happen on the ThreadPool. But when DisposeAsync throws, the observer has either unsubscribed or been notified of stream termination already, right? So you can't forward.

---

The enumerator can be diposed before, completion is pushed. As it was done in my first atempt. Then the catched exception can be forwarded.

---

FWIW, the sync counterpart doesn't attempt to catch anything: https://github.com/dotnet/reactive/blob/f650dded4d6d6ced6178f864f2eaecdbb57d5b43/Rx.NET/Source/src/System.Reactive/Linq/Observable/ToObservable.cs#L77

---

Exceptions for lifecycle operations such as `Get[Async]Enumerator` and `Dispose[Async]` can escape into the aether. Especially the latter, where exceptions during `Dispose` are different from exceptions due to retrieving values, and the two shouldn't be confused.

For the former of `Get[Async]Enumerator` there's often deferred propagation due to the way iterators work, i.e. an iterator enumerating over a sequence will see the call to the sequence's `Get[Async]Enumerator` occur in the iterator's first call to `MoveNext[Async]`, so exceptions caan naturally propagate there, because there's still a consumer and it's really an error in retrieving the data (i.e. not being able to start retrieving the data to begin with).

For `Dispose[Async]` things can be a bit more blurry, but a key observation is that we may be disposing because we reached an error state and called `observer.OnError` with such an error already, or because we have completed successfully and have called `observer.OnCompleted`. In both cases, the observer is in a terminal state and expects no further calls, so we can't bark an exception up its tree anymore either. Thus, exceptions from `Dispose[Async]` can just propagate. If the call to `Dispose[Async]` happens due to disposal of the operator, the exception may propagate to the caller. But if we're in some place removed from the user - e.g. on a threadpool - we have to let go of the exception there. In Rx, schedulers would enter the picture, with support for a `Catch` operator on schedulers, or global excpetion handler events on concrete scheduler types (akin to the thread pool having such handlers).